### PR TITLE
Add UnnecessaryBracesAroundTrailingLambda rule

### PR DESCRIPTION
**Pull request by atulgpt**
Fixes https://github.com/detekt/detekt/issues/6008

### CODE REVIEW

File: detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/movelambdaout/MoveLambdaOutsideParenthesesInspection.kt [hunk 1]
```diff
@@ -0,0 +1,81 @@
+// Copyright 2000-2022 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
```
Comment (BraisGabin) [inline detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/movelambdaout/MoveLambdaOutsideParenthesesInspection.kt:1]: I would remove this line. Below we are giving credits to the original source.

**Review by BraisGabin — APPROVED** [review]
Looks good!

File: detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/movelambdaout/MoveLambdaOutsideParenthesesInspection.kt (hunk 1 above)
Comment (atulgpt) [reply detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/movelambdaout/MoveLambdaOutsideParenthesesInspection.kt]: Done

File: detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/movelambdaout/UnnecessaryBracesAroundTrailingLambda.kt [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import io.gitlab.arturbosch.detekt.api.Severity
+import io.gitlab.arturbosch.detekt.api.internal.RequiresTypeResolution
+import org.jetbrains.kotlin.psi.KtCallExpression
+
+/**
+ * In Kotlin functions the last lambda parameter of a function is a function then a lambda expression passed as the
+ * corresponding argument can be placed outside the parentheses.
+ * see [Passing trailing lambdas](https://kotlinlang.org/docs/lambdas.html#passing-trailing-lambdas).
+ * Prefer the usage of trailing lambda.
+ */
```
Comment (cortinico) [inline detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/movelambdaout/UnnecessaryBracesAroundTrailingLambda.kt:18]: can we add `<compliant>` and `<noncompliant>` blocks?

Comment (atulgpt) [reply detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/movelambdaout/UnnecessaryBracesAroundTrailingLambda.kt:35]: Added

**Review by cortinico — APPROVED** [review]

**Pull request merged**